piwik_sdk/src/main/java/org/piwik/sdk/TrackMe.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Android SDK for Piwik
+ *
+ * @link https://github.com/piwik/piwik-android-sdk
+ * @license https://github.com/piwik/piwik-sdk-android/blob/master/LICENSE BSD-3 Clause
+ */
+
+package org.piwik.sdk;
+
+import android.support.annotation.NonNull;
+
+import java.util.HashMap;
+
+/**
+ * This objects represents one query to Piwik.
+ * For each event send to Piwik a TrackMe gets created, either explicitly by you or implicitly by the Tracker.
+ */
+public class TrackMe {
+    private static final int DEFAULT_QUERY_CAPACITY = 14;
+    private final HashMap<String, String> mQueryParams = new HashMap<>(DEFAULT_QUERY_CAPACITY);
+    private final CustomVariables mScreenCustomVariable = new CustomVariables();
+
+    /**
+     * You can set any additional Tracking API Parameters within the SDK.
+     * This includes for example the local time (parameters h, m and s).
+     * <pre>
+     * set(QueryParams.HOURS, "10");
+     * set(QueryParams.MINUTES, "45");
+     * set(QueryParams.SECONDS, "30");
+     * </pre>
+     *
+     * @param key   query params name
+     * @param value value
+     * @return tracker instance
+     */
+    public synchronized TrackMe set(@NonNull QueryParams key, String value) {
```

Comment:
It's an improvement, but not a solution.

Specifically the case of a developer handeling the TrackMe object between two threads, and the developer modifying the TrackMe object after passing it to the Tracker are problems.

Synchronizing the methods against the class was my idea to add some protection against that.
* While doing trySet it will never overwrite a value.
* While doing build the custom variables can not change.
* HashMap is not thread-safe
* CustomVariables are not threadsafe, we don't want them changing when putting them into the params in build()

But I'm not sure now how good this solution is though now you mentioned it.... The developer should not get ideas to try stuff like that anyways ;). If those methods can throw ConcurrencyModificationExceptions, they now can't.
We could go for a ConcurrentHashMap and less locking on the methods as potential improvement, while also making CustomVariables thread-safe.
Performance wise the synchronize statements can be neglected as TrackMe should not be accessed by multiple threads, but i agree that it's not a very clean approach.